src/main/java/com/mixpanel/android/mpmetrics/InAppFragment.java

```diff
@@ -148,11 +152,18 @@ public boolean onSingleTapUp(MotionEvent event) {
                     try {
                         Intent viewIntent = new Intent(Intent.ACTION_VIEW, uri);
                         mParent.startActivity(viewIntent);
-                        mMixpanel.getPeople().trackNotification("$campaign_open", inApp);
                     } catch (ActivityNotFoundException e) {
                         MPLog.i(LOGTAG, "User doesn't have an activity for notification URI " + uri);
                     }
+
+                    try {
+                        trackingProperties = new JSONObject();
+                        trackingProperties.put("url", uriString);
+                    } catch (final JSONException e) {
+                        MPLog.e(LOGTAG, "Can't put url into json properties");
+                    }
                 }
+                mMixpanel.getPeople().trackNotification("$campaign_open", inApp, trackingProperties);
```

Comment:
do we still want to track campaign open even if the mini in-app does nothing?

Reply:
i think so. it tracks taps, same as normal inapps